app/models/mail_server_log.rb

```diff
@@ -227,15 +227,18 @@ def delivery_status
       # attempt to parse the status from the log line and store if successful
       decorated = line(:decorate => true)
       if decorated && decorated.delivery_status
-        self.delivery_status = DeliveryStatusSerializer.
-                                 load(decorated.delivery_status)
-        delivery_status
```

Comment:
and wasn't needed anyway as the result of an assignment is the assigned value so we can just return that. Simplify!